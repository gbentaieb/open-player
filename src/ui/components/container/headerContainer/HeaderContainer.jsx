import React from 'react';
import { connect } from 'react-redux';
import HeaderWrapper from '../../presentational/headerWrapper/HeaderWrapper';
import ForcedMuttedButton from '../../presentational/forcedMuttedButton/ForcedMuttedButton';
import { setForcedMuted } from '../../../actions/CoreActions';

export function mapStateToProps(state) {
  return {
    forcedMuted: state.core.forcedMuted,
  };
}

const HeaderContainer = props => (
  <HeaderWrapper>
    <ForcedMuttedButton {...props} />
  </HeaderWrapper>
);


export default connect(
  mapStateToProps,
  {
    setForcedMuted,
  },
)(HeaderContainer);
