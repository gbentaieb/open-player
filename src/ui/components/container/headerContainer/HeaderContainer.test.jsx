import React from 'react';
import { createMockStore } from 'redux-test-utils';

import HeaderContainer, { mapStateToProps } from './HeaderContainer';

const defaultStore = {
  core: {
    forcedMuted: false,
  },
};

describe('HeaderContainer', () => {
  test('HeaderContainer > Snapshot', () => {
    const store = createMockStore(defaultStore);
    const wrapper = shallow(<HeaderContainer store={store} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('HeaderContainer > mapStateToProps', () => {
    const forcedMuted = false;
    const state = {
      core: { forcedMuted },
    };

    expect(mapStateToProps(state)).toEqual({
      forcedMuted,
    });
  });
});
