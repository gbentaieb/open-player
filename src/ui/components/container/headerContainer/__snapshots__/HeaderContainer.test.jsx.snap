// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer HeaderContainer > Snapshot 1`] = `
<HeaderContainer
  forcedMuted={false}
  setForcedMuted={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": undefined,
    }
  }
/>
`;
